distributed/client.py

```diff
@@ -3057,11 +3057,24 @@ def upload_file(self, filename, **kwargs):
         into a temporary directory on Python's system path so any .py,  .egg
         or .zip  files will be importable.
 
+        We recommend using :class:`distributed.diagnostics.plugin.UploadFile`
+        or :class:`distributed.diagnostics.plugin.UploadDirectory` when using
+        a dynamic cluster, to ensure that files are provided to new workers
+        as they join.
```

Comment:
Nitpick: `Client.upload_file` was refactored to use `UploadFile` internally, so files will also be uploaded to new workers when using `Client.upload_file`. This comment makes it seem like that's not the case though.